source/h2_stream.c

```diff
@@ -629,6 +685,14 @@ int aws_h2_stream_encode_data_frame(
         return AWS_OP_SUCCESS;
     }
 
+    /* EOF/complete on input stream for manual writes just means move to the next one, the body
+     * never truly completes until the stream is closed
```

Comment:
the input stream or the h2 stream? 

It looks like `body_complete` is always set to `false` for a "manual" write stream. I probably just missed it but where are we able to signal that the stream is truly done sending data? or does it not matter in practice?

Reply:
this is the input stream.

It does not matter at all that we signal manually that the stream is done, because closing it will take care of that. Closing a stream takes care of closing off any open frames and sends the final frame to the server.